src/ServicePulse.Host/vue/src/composables/serviceFailedMessage.ts

```diff
@@ -23,10 +23,15 @@ export async function useRetryEditedMessage(
   id: string,
   editedMessage: Ref<{
     messageBody: string;
-    headers: unknown[];
+    headers: any[];
   }>
 ) {
-  const payload = { message_body: editedMessage.value.messageBody, message_headers: editedMessage.value.headers };
+  let editedHeaders: { [key: string]: string } = {};
+  for (let index = 0; index < editedMessage.value.headers.length; index++) {
+    const header = editedMessage.value.headers[index];
+    editedHeaders[header.key] = header.value;
+  }
+  const payload = { message_body: editedMessage.value.messageBody, message_headers: editedHeaders };
```

Comment:
```suggestion
  let editedHeaders: { [key: string]: string } = {};
  for (let index = 0; index < editedMessage.value.headers.length; index++) {
    const header = editedMessage.value.headers[index];
    editedHeaders[header.key] = header.value;
  }
  const payload = { message_body: editedMessage.value.messageBody, message_headers: editedHeaders };
```
const payload = {
    message_body: editedMessage.value.messageBody,
    message_headers: editedMessage.value.headers.reduce((result, header) => {
      const { key, value } = header as { key: string; value: string };
      result[key] = value;
      return result;
    }),
  };

Reply:
this is going to need to be conditionally done according to the ServiceControl version. 4.33 works with the old headers array

Reply:
@PhilBastian from what I gather from the discussion, the old code used to pass extra props (which were ignored by JsonNet at deserialization time in SC), so all this code is doing is ensure those extra props are not being passed in, which as far as I understand is backwards compatible with prior versions of SC.
So I don't think we need to conditionally protect this code/

Reply:
old code sent an array to the API, containing objects with key and value properties. This code sends an object to the API, containing properties where the property names are the keys